internal/venice-common/src/main/java/com/linkedin/venice/controller/converters/ConverterUtil.java

```diff
@@ -0,0 +1,57 @@
+package com.linkedin.venice.controller.converters;
+
+import static com.linkedin.venice.controllerapi.ControllerApiConstants.*;
+
+import com.linkedin.venice.controller.requests.ControllerHttpRequest;
+import com.linkedin.venice.controller.requests.CreateStoreRequest;
+import com.linkedin.venice.controllerapi.NewStoreResponse;
+import com.linkedin.venice.controllerapi.QueryParams;
+import com.linkedin.venice.protocols.CreateStoreGrpcRequest;
+import com.linkedin.venice.protocols.CreateStoreGrpcResponse;
+import java.util.Optional;
+
+
+/**
+ * A utility class to hold all the conversion logic from {@link com.linkedin.venice.controller.requests.ControllerRequest}
+ * to transport specific request and transport specific responses back to {@link com.linkedin.venice.controllerapi.ControllerResponse}.
+ * The utility methods are registered with {@link GrpcConvertersRegistry} and {@link HttpConvertersRegistry} for the
+ * runtime to perform appropriate conversions. Refer to the documentation of the above converter registries for more
+ * information.
+ */
+public final class ConverterUtil {
+
+  // store related converters
+  public static CreateStoreGrpcRequest convertCreateStoreToGrpcRequest(CreateStoreRequest request) {
```

Comment:
Can this code be a function of `CreateStoreRequest`? Perhaps the parent class `ControllerRequest` can become `ControllerRequest<GRPC>` and define a `public abstract GRPC getGRPC()` API. Then `CreateStoreRequest extends ControllerRequest<CreateStoreGrpcRequest>`? Would that eliminate the need for the whole registry?

Reply:
This goes back to my earlier comment - https://github.com/linkedin/venice/pull/1259#discussion_r1884316176

Controller models (responses and requests) taking dependencies on the underlying transport through methods like `toGrpc` and `fromGrpc` would make the consumers of these models exposed to some underlying details (although we could control the access modifier) this also potential brings in the dependencies of the underlying transport dependencies for consumers of models.

Hence the choice to invert the dependency and have these concrete transport implementations depend on the models of controller abstraction and the conversion residing outside these models.